### PR TITLE
Dependabot: add CSP SDKs to the `allow` list

### PR DESCRIPTION
**Pull request by zmoog**
## Proposed commit message

- Add cloud provider SDKs to the `allow` list. 
- Focus GCP SDK packages to `cloud.google.com/go` only.

The `patterns` list in the `groups` section only matches the dependencies included in the `allow` list. To learn more, see https://github.com/elastic/beats/issues/39492#issuecomment-2217791699

The `cloud.google.com/go` packages focus on Google Cloud Platform (GCP) specifically, allowing the use of APIs for services like Compute Engine, Cloud Storage, BigQuery, Pub/Sub, and others.

## Checklist

- [ ] ~~My code follows the style guidelines of this project~~
- [ ] ~~I have commented my code, particularly in hard-to-understand areas~~
- [ ] ~~I have made corresponding changes to the documentation~~
- [ ] ~~I have made corresponding change to the default configuration files~~
- [ ] ~~I have added tests that prove my fix is effective or that my feature works~~
- [ ] ~~I have added an entry in `CHANGELOG.next.asciidoc` or `CHANGELOG-developer.next.asciidoc`.~~

## Related issues

- Relates #39492

### CODE REVIEW

**Comment by zmoog** [comment]
Hey @dliappis, my apologies for submitting yet another PR on Dependabot 🤦 

It seems the `groups` section only matches the dependencies included in the `allow` list. By default, `allow: *`, but since in Beats we have an explicit `allow` list, I need to add the `dependency-name` as well.

**Comment by zmoog** [comment]
Hey @dliappis, could you look at this PR that updates the Dependabot config? This config is (probably) a good one!

File: .github/dependabot.yml [hunk 1]
```diff
@@ -35,18 +35,23 @@ updates:
...
         patterns:
-          - "github.com/Azure/azure-sdk-for-go/*"
-          - "github.com/Azure/azure-event-hubs-go/*"
+          - "github.com/Azure/azure-*-go"
           - "github.com/Azure/go-autorest/*"
-          - "github.com/Azure/azure-storage-blob-go/*"
       gcp-sdks:
         patterns:
           - "cloud.google.com/go/*"
-          - "google.golang.org/*"
```
Comment (dliappis) [inline .github/dependabot.yml:49]: Just double checking: this is on purpose?

The namespace at least seems to exist https://pkg.go.dev/google.golang.org/api

**Review by dliappis — COMMENTED** [review]
Looks good, left a small question

File: .github/dependabot.yml (hunk 1 above)
Comment (zmoog) [reply .github/dependabot.yml:49]: The `cloud.google.com/go/*` packages focus on Google Cloud Platform (GCP) specifically, allowing the use of APIs for services like Compute Engine, Cloud Storage, BigQuery, Pub/Sub, and others.

The `https://pkg.go.dev/google.golang.org/api` packages are applicable instead for interacting with various Google APIs, such as Google Drive, Google Calendar, YouTube, and others.

For this Dependabot configuration, I want to focus only on GCP. My bad for not adding a comment in the PR description; I updated it accordingly).

Comment (zmoog) [reply .github/dependabot.yml:49]: @dliappis, let me know what you think 🙇

Comment (rowlandgeoff) [reply .github/dependabot.yml:49]: @zmoog Thanks for the explanation. I will approve the PR on behalf of @dliappis since he isn't available this week.

**Review by rowlandgeoff — APPROVED** [review]

File: .github/dependabot.yml (hunk 1 above)
Comment (zmoog) [reply .github/dependabot.yml:49]: Thank you, @rowlandgeoff!

**Pull request merged**